cmd/dep/init_test.go

```diff
@@ -33,3 +37,46 @@ func TestIsStdLib(t *testing.T) {
 		}
 	}
 }
+
+func TestGetProjectPropertiesFromVersion(t *testing.T) {
+	cases := []struct {
+		version gps.Version
+		want    gps.Version
+	}{
+		{
+			version: gps.NewBranch("foo-branch"),
+			want:    gps.NewBranch("foo-branch"),
+		},
+		{
+			version: gps.NewVersion("foo-version"),
+			want:    gps.NewVersion("foo-version"),
+		},
+		{
+			version: gps.NewBranch("foo-branch").Is("some-revision"),
+			want:    gps.NewBranch("foo-branch"),
+		},
+		{
+			version: gps.NewVersion("foo-version").Is("some-revision"),
+			want:    gps.NewVersion("foo-version"),
+		},
+		{
+			version: gps.Revision("some-revision"),
+			want:    nil,
+		},
+	}
+
+	for _, c := range cases {
+		actualProp := getProjectPropertiesFromVersion(c.version)
+		if c.want != actualProp.Constraint {
+			t.Fatalf("Expected project property to be %v, got %v", actualProp.Constraint, c.want)
+		}
+	}
+
+	// Test to have caret in semver version
+	outsemver := getProjectPropertiesFromVersion(gps.NewVersion("v1.0.0"))
+	wantSemver, _ := gps.NewSemverConstraint("^1.0.0")
```

Comment:
The actual problem here is an annoying implementation detail - it should work without needing to fall back on string comparison if you use `^v1.0.0` instead of `^1.0.0`. I believe this kind of issue is improved in newer versions of the semver lib, but we're stuck with what we've got for now.